label_studio/io_storages/base_models.py

```diff
@@ -341,17 +342,20 @@ def _scan_and_create_links_v2(self):
         raise NotImplementedError
 
     @classmethod
-    def add_task(cls, data, project, maximum_annotations, max_inner_id, storage, key, row_index, link_class):
+    def add_task(cls, project, maximum_annotations, max_inner_id, storage, link_params, link_class):
+        link_kwargs = asdict(link_params)
+        data = link_kwargs.pop('task_data')
```

Comment:
is there a risk of a crash if the key is missing from the dict? genuinely asking, not extremely familiar with dataclasses

Reply:
nope, at this point it's a regular dict and `.pop` has an implicit default `None`

Reply:
It's counterintuitive next to `dict#get`, but:

```
>>> {}.pop('foo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'foo'
```

you have to pass `pop()` the `None` default if that's the expected behavior.

Reply:
oops, you're right!

Reply:
I don't think the dataclass could have had it missing to begin with, but good if someone accidentally passes another kind of object in there